pkg/pxc/app/volumes.go

```diff
@@ -0,0 +1,48 @@
+package app
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	api "github.com/Percona-Lab/percona-xtradb-cluster-operator/pkg/apis/pxc/v1alpha1"
+)
+
+func GetConfigVolumes(cr *api.PerconaXtraDBCluster, cvName string) corev1.Volume {
+	vol1 := corev1.Volume{
+		Name: "config-volume",
+	}
+
+	vol1.ConfigMap = &corev1.ConfigMapVolumeSource{}
+	vol1.ConfigMap.Name = cvName
+	t := true
+	vol1.ConfigMap.Optional = &t
+	return vol1
+}
+
+func GetDataVolumes(cr *api.PerconaXtraDBCluster, sfs api.StatefulApp, dataVolumeName string) (bool, []corev1.PersistentVolumeClaim, corev1.VolumeSource) {
```

Comment:
I'm not sure we need this function at all.
I see that you wanted to avoid code copying in `Volumes` methods. But adding this abstraction you eventually ended up with copy-pasting of different code. I mean:
```go
var volume api.Volume
if isPVC {
    volume.PVCs = pvcs
} else {
    volume.Volumes = []corev1.Volume{
        corev1.Volume{
            VolumeSource: dataVolume,
            Name:         proxyDataVolumeName,
        },
    }
}
```
All in all there is just more code and clumsy constructions as `isPVC` etc.

Just move this checks to `Volumes` methods and ditch this function.